### ISSUE TITLE
"Creating new version failed: Git tag not found" although tag is present

### ISSUE THREAD

**Issue by shroudedcode** [atom/apm#757]
### Prerequisites

* [x] Put an X between the brackets on this line if you have done all of the following [...]

### Description

I'm currently [working on a @semantic-release plugin for apm](https://github.com/shroudedcode/semantic-release-apm/issues/4#issuecomment-347038294) that has to publish a package based on an existing tag set by it. Whatever I try I don't seem to get it to work, `apm publish --tag vX.Y.Z` throws an error saying the tag doesn't exist although it clearly does.

https://asciinema.org/a/MT5yNuRehajpwPGkJXJ1OXu2X

### Steps to Reproduce

- Create an Atom package.
- Login to APM with your token.
- Follow what I did in [this asciinema recording](https://asciinema.org/a/MT5yNuRehajpwPGkJXJ1OXu2X) (note that I tried this multiple times and with longer delays).

**Expected behavior:**

The package gets successfully published.

**Actual behavior:**

```
Publishing semantic-release-apm-example@vX.Y.Z ✗
Creating new version failed: Git tag not found
```

**Reproduces how often:** 100%

### Versions

```
Atom    : 1.22.1
Electron: 1.6.15
Chrome  : 56.0.2924.87
Node    : 7.4.0
apm  1.18.8
npm  3.10.10
node 6.9.5 x64
python 2.7.12
git 2.7.4
```
on `Ubuntu 16.04.3 LTS`

**Comment by rsese** [comment]
Sorry for the trouble @shroudedcode - it looks like the issue is with https://github.com/shroudedcode/semantic-release-apm-example yes?    Can you confirm if you published that package and then deleted/re-created the repository?  I'm wondering if you're hitting the situation described in https://github.com/atom/apm/issues/707#issuecomment-295773128.

**Comment by shroudedcode** [comment]
> Can you confirm if you published that package and then deleted/re-created the repository?

Yeah ... seems like I re-created the repository at some point while trying things out.
Totally didn't think of that (because I wasn't aware of repository IDs).

> I'm wondering if you're hitting the situation described in #707 (comment).

Looks very similar, I also tried the suggested fix (`unpublish` and `publish` again) and it worked.

---

While my issue is now solved, I think that you should add more telling error messages to prevent this from happening in the future, something like this:

```
The repository your package is linked to no longer exists. If you re-created the
repository at some point, try updating the repository link with "apm relink".
```

This way it's clear that nothing is wrong with the tag/tarball, but that the wrong repository is linked.

**Comment by kachkaev** [comment]
I was desperately trying to publish a [fork](https://github.com/gicentre/markdown-preview-enhanced-with-litvis) of [markdown-preview-enhanced](https://github.com/shd101wyy/markdown-preview-enhanced) now and wasted more than an hour. Here's what I saw no matter what options etc. I tried:

```
apm publish minor
Preparing and tagging a new version ✓
Pushing v0.2.0 tag ✓
Publishing markdown-preview-enhanced-with-litvis@v0.2.0 ✗
Creating new version failed: Git tag not found
```

```
apm publish --tag v0.2.0
Publishing markdown-preview-enhanced-with-litvis@v0.2.0 ✗
Creating new version failed: Git tag not found
```

After I did all possible sane and insane things such as adding or removing `.git` in `package.json`'s `repository` field, manually pushing to master while the update was going, tweaking `.git/config`, etc, I noticed that in my fork I also had the tags from the original repo. So I was publishing `v0.2.0` while there were tags like `v0.3.0` ... `v0.15.5` in the repo. Wiping out these tags on github either helped or was just co-incised with a successful publish. Nevermind, going to sleep now 😫
